build/bazel/remote/logstream/v1/remote_logstream.proto

```diff
@@ -0,0 +1,139 @@
+// within a reasonable period of time (implementation-defined). Committed bytes
+// to a LogStream cannot be overwritten, and finalized LogStreams - indicated by
+// setting `finish_write` field in the final WriteRequest - also cannot be
+// appended to.
+//
+// When calling the Byte Stream API's `Write` RPC to write LogStreams, writers
+// must pass the `write_resource_name` of a LogStream as
+// `ByteStream.WriteRequest.resource_name` rather than the LogStream's `name`.
+// Separate resource names for reading and writing allows for broadcasting the
+// read resource name widely while simultaneously ensuring that only writer(s)
+// with knowledge of the write resource name may have written bytes to the
+// LogStream.
+//
+// #### Reading LogStreams
+//
+// Use the Byte Stream API's `Read` RPC to read LogStreams. When reading
+// finalized LogStreams the server will stream all contents of the LogStream
+// starting at `ByteStream.ReadRequest.read_offset`.
+//
+// When reading unfinalized LogStreams the server must keep the streaming
+// `ByteStream.Read` RPC open and send `ByteStream.ReadResponse` messages as
+// more bytes become available or the LogStream is finalized.
+//
+// #### Example Multi-Party Read/Write Flow
+//
+// 1. LogStream Writer calls `CreateLogStream`
+// 2. LogStream Writer publishes `LogStream.name`
+// 3. LogStream Writer calls `ByteStream.Write` with
+//    `LogStream.write_resource_name` as
+//    `ByteStream.WriteRequest.resource_name`,
+//    `ByteStream.WriteRequest.finish_write`=false.
+// 4. LogStream Reader(s) call `ByteStream.Read` with the published
+//    `LogStream.name` as `ByteStream.ReadRequest.resource_name`.
+// 5. LogStream Service streams all committed bytes to LogStream Reader(s),
+//    leave the stream open.
+// 6. LogStream Writer calls `ByteStream.Write` with
+//    `LogStream.write_resource_name` as
+//    `ByteStream.WriteRequest.resource_name`,
+//    `ByteStream.WriteRequest.finish_write`=true.
+// 7. LogStream Service streams all remaining bytes to LogStream Reader(s),
+//    terminates the stream.
+service LogStreamService {
+  // Create a LogStream which may be written to.
+  //
+  // The returned LogStream resource name will include a `write_resource_name`
+  // which is the resource to use when writing to the LogStream.
+  // Callers of CreateLogStream are expected to NOT publish the
+  // `write_resource_name`.
+  rpc CreateLogStream(CreateLogStreamRequest) returns (LogStream) {}
+}
+
+// Contains all information necessary to create a new LogStream resource.
+message CreateLogStreamRequest {
+  // Required. The parent resource of the created LogStream.
+  // The list of valid types of parent resources of LogStreams is up to the
+  // implementing server.
+  // Example: projects/123
+  string parent = 1;
+}
+
+// A handle to a log (an ordered sequence of bytes).
+message LogStream {
+  // Structured name of the resource in the format:
+  //   {parent=**}/logStreams/{logstream_id}
```

Comment:
Let's correct this while we're here: logStreams -> logstreams. (Resources are usually lowercase, and this is already wrong with having a lowercase l and uppercase S)

Reply:
Couple more: lines 114, 120, and 121 also thanks!

Reply:
Ah good catch! Should be fixed now.